packages/api/amplify_api_dart/lib/src/util/amplify_api_config.dart

```diff
@@ -37,7 +37,7 @@ class EndpointConfig with AWSEquatable<EndpointConfig> {
     return parsed.replace(
       pathSegments: [
         ...parsed.pathSegments,
-        ...pathSegmentsFromPath,
+        if (pathSegmentsFromPath.isNotEmpty && pathSegmentsFromPath.first.isNotEmpty) ...pathSegmentsFromPath,
       ],
```

Comment:
Would this be better?

```dart
pathSegments: [
  ...parsed.pathSegments,
  ...pathSegmentsFromPath,
].where((segment) => segment.isNotEmpty),
```

Since this method is shared between REST and GraphQL, could there be use cases where a trailing slash is desired?

Reply:
Yes this could be better and has the same effect. But unfortunately it doesn't solve our problem.

Reply:
I have updated the PR with your suggested code. We test our code and this fix works fine!